modules/affile/affile-dest.c

```diff
@@ -316,6 +316,14 @@ affile_dw_set_owner(AFFileDestWriter *self, AFFileDestDriver *owner)
                            self->filename);
 }
 
+static void
+affile_dw_unset_owner(AFFileDestWriter *self)
+{
+  if (self->owner)
+    log_pipe_unref(&self->owner->super.super.super);
+  self->owner = NULL;
```

Comment:
Probably we should do something with `self->super.expr_node` too.

Reply:
Upon a closer look it seems that the `expr_node` setting is done without freeing the old one in other places as well.

Should I fix it in this PR or in a separate one?